src/java/edu/slu/action/ObjectAction.java

```diff
@@ -793,73 +831,82 @@ public void updateObject() throws IOException, ServletException, Exception{
-                    catch (IOException ex) {
-                        Logger.getLogger(ObjectAction.class.getName()).log(Level.SEVERE, null, ex);
+                    newObject = configureRerumOptions(newObject, true); //__rerum for the new object being created because of the update action
+                    newObject.remove("@id"); //This is being saved as a new object, so remove this @id for the new one to be set.
+                    //Since we ignore changes to __rerum for existing objects, we do no configureRerumOptions(updatedObject);
+                    DBObject dbo = (DBObject) JSON.parse(newObject.toString());
+                    String newNextID = mongoDBService.save(Constant.COLLECTION_ANNOTATION, dbo);
+                    String newNextAtID = "http://devstore.rerum.io/rerumserver/id/"+newNextID;
+                    BasicDBObject dboWithObjectID = new BasicDBObject((BasicDBObject)dbo);
+                    dboWithObjectID.append("@id", newNextAtID);
+                    newObject.element("@id", newNextAtID);
+                    mongoDBService.update(Constant.COLLECTION_ANNOTATION, dbo, dboWithObjectID);
+                    historyNextUpdatePassed = alterHistoryNext(updateHistoryNextID, newNextAtID); //update history.next or original object to include the newObject @id
+                    if(historyNextUpdatePassed){
+                        JSONObject jo = new JSONObject();
+                        JSONObject iiif_validation_response = checkIIIFCompliance(newNextAtID, "2.1");
+                        jo.element("code", HttpServletResponse.SC_OK);
+                        jo.element("original_object_id", updateHistoryNextID);
+                        jo.element("new_obj_state", newObject); //FIXME: @webanno standards say this should be the response.
+                        jo.element("iiif_validation", iiif_validation_response);
+                        try {
+                            addWebAnnotationHeaders(newNextID, isContainerType(newObject), isLD(newObject));
+                            response.addHeader("Access-Control-Allow-Origin", "*");
+                            response.setStatus(HttpServletResponse.SC_OK);
+                            out = response.getWriter();
+                            out.write(mapper.writer().withDefaultPrettyPrinter().writeValueAsString(jo));
+                        } 
+                        catch (IOException ex) {
+                            Logger.getLogger(ObjectAction.class.getName()).log(Level.SEVERE, null, ex);
+                        }
+                    }
+                    else{
+                        //The error is already written to response.out, do nothing.
                     }
                 }
                 else{
-                    //The error is already written to response.out, do nothing.
+                    writeErrorResponse("Object "+received.getString("@id")+" not found in RERUM, could not update.", HttpServletResponse.SC_BAD_REQUEST);
```

Comment:
This case is actually the remote object case, isn't it? Shouldn't we create and add the @id to __rerum.history.previous?

Reply:
We need to break this out into its own ticket.  We need to rethink what update.action does and how to separate and handle PUT vs PATCH gracefully and compliantly.